### PR TITLE
feat(x/ecocredit): `Cancel` ORM

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

adds cancel method with ORM

ref: #696 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/cancel.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	"context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/types"
+	"github.com/regen-network/regen-ledger/types/math"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+func (k Keeper) Cancel(ctx context.Context, req *v1beta1.MsgCancel) (*v1beta1.MsgCancelResponse, error) {
```
Comment (ryanchristo) [inline x/ecocredit/server/core/cancel.go:12]: godoc :pray:

File: x/ecocredit/server/core/cancel.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	sdkCtx := types.UnwrapSDKContext(ctx)
+	holder, _ := sdk.AccAddressFromBech32(req.Holder)
```
Comment (ryanchristo) [inline x/ecocredit/server/core/cancel.go:14]: check error :pray:

File: x/ecocredit/server/core/cancel_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+	sup, err := s.stateStore.BatchSupplyStore().Get(s.ctx, 1)
+	assert.NilError(t, err)
+	assert.Equal(t, sup.TradableAmount, "0.0")
+	assert.Equal(t, sup.RetiredAmount, "10.5")
+
+	bal, err := s.stateStore.BatchBalanceStore().Get(s.ctx, s.addr, 1)
+	assert.NilError(t, err)
+	assert.Equal(t, bal.Tradable, "0.0")
+	assert.Equal(t, bal.Retired, "10.5")
```
Comment (ryanchristo) [inline x/ecocredit/server/core/cancel_test.go:41]: Comments with original values and math would be nice.

**Review by ryanchristo — APPROVED** [review]
Looks good. A few minor comments. Pre-approving.

**Review by atheeshp — APPROVED** [review]

**Pull request merged**